.github/workflows/CD-pipeline-dev.yml

```diff
@@ -35,16 +38,15 @@ jobs:
           version: 3.6.1
 
       - name: Login to GCP
-        uses: google-github-actions/setup-gcloud@v0.6.0
+        uses: google-github-actions/auth@v0.4.0
         with:
-          project_id: ${{ secrets.GCP_PROJECT_ID }}
-          service_account_email: ${{ secrets.GCP_SERVICE_ACCOUNT_EMAIL }}
-          service_account_key: ${{ secrets.GCP_DOCKER_REGISTRY_KEY }}
-          export_default_credentials: true
-          cleanup_credentials: true
+          token_format: 'access_token'
+          workload_identity_provider: ${{ secrets.GCP_WIF_PROVIDER }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT_EMAIL }}
+          access_token_lifetime: '21600s'
 
       - name: Configure kubectl
-        uses: google-github-actions/get-gke-credentials@v0.3.0
+        uses: google-github-actions/get-gke-credentials@v0.4.0
         with:
           cluster_name: ${{ secrets.GKE_CLUSTER_NAME }}
           location: ${{ secrets.GKE_ZONE }}
```

Comment:
Please make 100% sure that the `deploy-and-monitor` pipeline is exactly equivalent (duplicate code) between CD and CI (please let me know if this doesn't make sense anymore).
I see some diffs, e.g. `credentials: ${{ secrets.GKE_SERVICE_ACC_KEY }}`

Reply:
Thanks for finding the disparities, i have gone through the changes again and have fixed the CD.